backend/sirius-web-collaborative-forms-api/src/main/java/org/eclipse/sirius/web/collaborative/forms/api/dto/UpdateWidgetFocusSuccessPayload.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.collaborative.forms.api.dto;
+
+import java.text.MessageFormat;
+import java.util.Objects;
+
+import org.eclipse.sirius.web.annotations.graphql.GraphQLField;
+import org.eclipse.sirius.web.annotations.graphql.GraphQLNonNull;
+import org.eclipse.sirius.web.annotations.graphql.GraphQLObjectType;
+import org.eclipse.sirius.web.services.api.dto.IPayload;
+
+/**
+ * The payload of the "Widget Focus" mutation returned on success.
+ *
+ * @author smonnier
+ */
+@GraphQLObjectType
+public final class UpdateWidgetFocusSuccessPayload implements IPayload {
+    private final UpdateWidgetFocusInput widget;
```

Comment:
Do not return the input in the payload. You do not need to have anything here, this UpdateWidgetFocusSuccessPayload does not need any field for now. Its presence as a response is ok.

Reply:
When I removed the input, I was not able to launch the odweb sample application anymore as it stopped after throwing exceptions starting with: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'webSocketConfiguration' defined in file [C:\Git\OCP\sirius-components\backend\sirius-web-spring-graphql\target\classes\org\eclipse\sirius\web\spring\graphql\configuration\WebSocketConfiguration.class]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'graphQL' defined in class path resource [org/eclipse/sirius/web/graphql/configuration/GraphQLConfiguration.class]: Unsatisfied dependency expressed through method 'graphQL' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'graphQLSchema' defined in class path resource [org/eclipse/sirius/web/graphql/configuration/GraphQLConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [graphql.schema.GraphQLSchema]: Factory method 'graphQLSchema' threw exception; nested exception is graphql.schema.validation.InvalidSchemaException: invalid schema:
"UpdateWidgetFocusSuccessPayload" must define one or more fields.

Reply:
In that case, return just the widgetId or something simple like that. It's odd that we can't have empty objects, it was supported before. I'll have to look into that.

Reply:
So after having a quick look at the GraphQL specification, it is indeed an error to have an empty object in the schema. So just return the widgetId to indicate which widget has been focused.

Reply:
Ok I replaced the widget in input by its widgetID